src/app/model/ChessEngine.ts

```diff
@@ -1,16 +1,18 @@
 import { ChessModel } from './ChessModel';
-import { Side, Square } from './Types';
+import { PiecesBoardPositions, Side, Square } from './Types';
 import { Piece } from './Piece';
 import { Player } from './Player';
-import { ChessBoard } from './ChessBoard';
+import { Chessboard } from './Chessboard';
 import { PieceWasMoved } from './PieceWasMoved';
 import { PieceWasCaptured } from './PieceWasCaptured';
 import { isDefined } from './HelperFunctions';
+import { PIECES_START_POSITION } from './Constances';
 
 export class ChessEngine implements ChessModel {
   private currentSide: Side = Side.BLACK;
+  piecesPositions: PiecesBoardPositions = PIECES_START_POSITION;
```

Comment:
To akurat ma mały sens. Skoro zmienna robisz, tylko poto, zeby ja odczytac i jeszcze w innym miejscu, to w ogole nie ma po co jej tutaj dawac. 
Hmmm... są 2 opcje: albo w presenterze w jakis sposob wyciagamy z chessEngine ustawienie pionków i mapujemy je na reprezentacje taką jaką chce widok wg. typu PiecesBoardPositions (to powininen być typ view - teraz mamy zaleności między view i model) albo zakladamy, ze startGame po prostu zawsze pokazuje poczatkowy uklad (proponuje takie rozwiazanie, skoro nie mamy wczytywania gry). I pomijamy zupelnie model. 

Założenie było takie.
1. View mowi, chce otrzymac ustawienie pionkow w typie PiecesBoardPositions (to jest typ mój i ja go rozumiem, a nie model). O modelu nawet nie wiem, ze istnieje.
2. Model mówi - ja wiem, jakie jest ułożenie pionków i możesz się tego dowiedzieć w sposób... (to zależy - np. iterując i sprawdzając onPositionPiece / albo jakos tworzac podobna strukture). O widoku nawet nie wiem, ze istnieje.
3. Presenter mowi: Ja wiem, ze istnieje widok, ktory mowi jaka strukture chce dostac do wyswietlenia. I wiem, ze istnieje model, ktory w jakis sposob dostarczy mi tych informacji, ktore chce widok. Moim zadaniem jest tak przetłumaczyc wiedze z modelu, żeby widok dostał w zrozumiały dla siebie sposób. To tak jakby widok mówił po polsku, model po chińksku a ja jestem tłumaczem tych języków. Sami w ogóle nie mogą ze sobą rozmawiać. 

Pomogło to :) ? Bo teraz zmieniłeś model na potrzeby widoku no i zrobiles, ze wiedza o sobie.